Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py

```diff
@@ -74,32 +79,49 @@ def get(self, number):
         def find(self, opened=True, head=None, base=None):
             assert opened in (True, False, None)
 
-            user, _ = self.repository.credentials()
-            qhead = ''
-            if head and ':' in head:
-                qhead = head
-            elif user and head:
-                qhead = '{}:{}'.format(user, head)
-
-            data = self.repository.request('pulls', params=dict(
-                state={
-                    None: 'all',
-                    True: 'open',
-                    False: 'closed',
-                }.get(opened),
-                base=base,
-                head=qhead,
-            ))
+            search = 'repo:{}/{} is:pr'.format(self.repository.owner, self.repository.name)
+            if head:
+                search += ' head:{}'.format(head)
+            if base:
+                search += ' base:{}'.format(base)
+            search += {
+                True: ' is:open',
+                False: ' is:closed',
+            }.get(opened, '')
+
+            data = self.repository.graphql('''query {{
+  search(query: "{}", type: ISSUE, last: 100) {{
```

Comment:
Right, the reason I originally did the pagination thing is because GitHub doesn't let you ask for a PR by `head` name unless the `head` name includes the user. For some reason, this API doesn't work reliably in private repositories (meaning we can't find an existing PR), but the graphQL API does work.

Reply:
In that case, should we expect the API to always return one match? What is the 100 limit based off?

Reply:
We can expect the API to return some matches, mostly likely 0 or 1, but if a branch name has been repeated, possibly more. Unlikely to be more than a dozen. If we need to support more interesting queries going forward, I'd sort of like for this testing function to be made a bit more robust. For now, the goal of this PR is to support a relatively specific use case involving a private repository.